src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -913,6 +927,43 @@ impl<'a, 'tcx> BorrowckCtxt<'a, 'tcx> {
         err.emit();
     }
 
+    fn immutable_argument_should_be_mut(&self, nid: ast::NodeId, db: &mut DiagnosticBuilder) {
+        let parent = self.tcx.map.get_parent_node(nid);
+
+        if let Some(fn_like) = FnLikeNode::from_node(self.tcx.map.get(parent)) {
```

Comment:
This is moved code from bellow so that I could use it twice without copying code around :)

I will see how I can make it easier to follow.